### PR TITLE
refactor(sarif): extract magic numbers and split buildCodeNarcRules

### LINKED ISSUE #922: [sarif] Extract magic numbers and split buildCodeNarcRules
## Problem

`SarifRuleRegistry.kt` has 23 detekt violations:
- 22 `MagicNumber` violations (priority values like `3`, `4`, `5`)
- 1 `LongMethod` (`buildCodeNarcRules` is 81 lines, max 60)

## Context

The magic numbers represent SARIF rule priorities and CodeNarc rule configurations. These should be extracted to named constants for clarity.

## Proposed Approach

1. **Extract priority constants:**
   ```kotlin
   private object SarifPriority {
       const val HIGH = 3
       const val MEDIUM = 4
       const val LOW = 5
       // etc.
   }
   ```

2. **Split `buildCodeNarcRules`** into smaller categorized functions:
   - `buildBasicRules()`
   - `buildFormattingRules()`
   - `buildNamingRules()`
   - etc.

## Test Coverage

✅ 6 existing tests in `SarifRuleRegistryTest.kt` - safe to refactor

## References

- File: `groovy-diagnostics/sarif/src/main/kotlin/.../SarifRuleRegistry.kt`
- Parent issue: #578

### PR DESCRIPTION
**Pull request by albertocavalcante**
## Summary

- Extract `CodeNarcPriority` constants (`HIGH=1`, `MEDIUM=2`, `LOW=3`) to eliminate magic numbers
- Split `buildCodeNarcRules()` (100+ lines) into 10 focused category helpers
- Add 17 comprehensive tests for edge cases, validation, and thread safety

## Detekt Violations Fixed

| Violation | Count | Resolution |
|-----------|-------|------------|
| `MagicNumber` | 22 | Replaced with `CodeNarcPriority.HIGH/MEDIUM/LOW` |
| `LongMethod` | 1 | Split into 10 categorized helper functions |

## Test Coverage Added

- **Edge cases**: empty string, whitespace, case sensitivity
- **Category coverage**: all 11 categories verified
- **URL validation**: helpUri format correctness
- **Invariants**: rule count guards, no duplicates, required fields
- **Priority mapping**: 1→ERROR, 2→WARNING, 3→NOTE
- **Thread safety**: concurrent access verification

## Test plan

- [x] All 23 tests pass (`./gradlew :groovy-diagnostics:sarif:test`)
- [x] Lint passes (`./gradlew lintFix`)
- [ ] CI passes

Closes #922

## Summary by CodeRabbit

* **New Features**
  * Expanded CodeNarc rule coverage from existing rules to 60+ categorized rules for improved diagnostics.
  * Enhanced severity mapping with consistent priority levels (High, Medium, Low) aligned to diagnostic standards.

* **Tests**
  * Added comprehensive validation tests for rule consistency, URL formatting, thread safety, and field requirements.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: groovy-diagnostics/sarif/src/test/kotlin/com/github/albertocavalcante/diagnostics/sarif/SarifRuleRegistryTest.kt [hunk 1]
```diff
@@ -100,4 +100,208 @@ class SarifRuleRegistryTest {
...
+    fun `compiler rules should have groovy-lang helpUri`() {
+        val rules = SarifRuleRegistry.getCompilerRules()
+
+        rules.forEach { rule ->
+            assertNotNull(rule.helpUri)
+            assertTrue(
+                rule.helpUri?.contains("groovy-lang.org") == true,
+                "Compiler rule ${rule.id} should have groovy-lang.org URI",
+            )
+        }
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-diagnostics/sarif/src/test/kotlin/com/github/albertocavalcante/diagnostics/sarif/SarifRuleRegistryTest.kt]: Fixed in aaa4f16b.

File: groovy-diagnostics/sarif/src/test/kotlin/com/github/albertocavalcante/diagnostics/sarif/SarifRuleRegistryTest.kt [hunk 2]
```diff
@@ -100,4 +100,208 @@ class SarifRuleRegistryTest {
...
+                }
+            }
+        }
+
+        threads.forEach { it.start() }
+        threads.forEach { it.join() }
+    }
+
+    @Test
+    fun `multiple calls to getCodeNarcRules should return same instances`() {
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-diagnostics/sarif/src/test/kotlin/com/github/albertocavalcante/diagnostics/sarif/SarifRuleRegistryTest.kt]: Fixed in aaa4f16b.

File: groovy-diagnostics/sarif/src/test/kotlin/com/github/albertocavalcante/diagnostics/sarif/SarifRuleRegistryTest.kt [hunk 3]
```diff
@@ -100,4 +100,208 @@ class SarifRuleRegistryTest {
...
+                assertTrue(priority >= 1, "Rule ${rule.id} has invalid priority $priority (should be >= 1)")
+            }
+        }
+    }
+
+    // Adversarial tests
+
+    @Test
+    fun `concurrent access to getRule should be thread safe`() {
+        val threads = (1..10).map { threadNum ->
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-diagnostics/sarif/src/test/kotlin/com/github/albertocavalcante/diagnostics/sarif/SarifRuleRegistryTest.kt]: Fixed in cf92ce39.
